File: fixtures/hub-cobol-clbs-mini/batch/reference_emit_idxngprn.java
// EXPECTED: 75.50
// Reference emit for IDXNGPRN — INDEXED START NOT GREATER + READ PREV.
import java.util.*;
public final class IdxngprnRef {
  public static void main(String[] args) {
    Map<Integer, Double> rows = new HashMap<>();
    rows.put(10, 12.50); rows.put(20, 8.00); rows.put(42, 25.00);
    rows.put(55, 30.00); rows.put(99, 15.00);
    int start = 55;
    double total = 0;
    for (Map.Entry<Integer, Double> e : rows.entrySet()) {
      if (e.getKey() <= start) total += e.getValue();
    }
    System.out.printf(Locale.US, "%.2f%n", total);
  }
}

File: fixtures/hub-cobol-clbs-mini/batch/reference_emit_idxngprn.py
"""Reference emit for IDXNGPRN — INDEXED START NOT GREATER + READ PREV → 75.50."""
# EXPECTED: 75.50
ROWS = {10: 12.50, 20: 8.00, 42: 25.00, 55: 30.00, 99: 15.00}
START = 55
# GnuCOBOL BDB: START NOT GREATER positions at greatest key <= START; PREV includes it.
total = sum(v for k, v in ROWS.items() if int(k) <= START)
print(f"{total:.2f}")
